drivers/dma/dma_mcux_lpc.c

```diff
@@ -193,39 +200,110 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 				config->block_count);
 			return -EINVAL;
 		}
-		int i = 0;
-		dma_transfer_config_t *next_transfer;
+
+		/* Allocate the descriptor table structures if needed */
+		if (data->dma_descriptor_table == NULL) {
+			data->dma_descriptor_table = k_malloc(CONFIG_DMA_LINK_QUEUE_SIZE *
+							(sizeof(dma_descriptor_t) +
+							 FSL_FEATURE_DMA_LINK_DESCRIPTOR_ALIGN_SIZE));
+
+			if (!data->dma_descriptor_table) {
+				LOG_ERR("HEAP_MEM_POOL_SIZE is too small");
+				return -ENOMEM;
+			}
+			/* Ensure descriptor is aligned */
+			data->dma_descriptor_table = (dma_descriptor_t *)ROUND_UP(
```

Comment:
if we need free the memory block later, then we need keep the old memory pointor

Reply:
Good point, I will update the driver to watch for this.